test/models/stacks_test.rb

```diff
@@ -591,5 +591,12 @@ def setup
       assert_equal new_reason, @stack.lock_reason
       assert_equal old_time, @stack.locked_since
     end
+
+    test "stack should contains valid deploy_url" do
+      @stack.deploy_url = "Javascript:alert(0);//"
+      assert_not @stack.valid?
```

Comment:
I think both are valid in a `ActiveSupport::TestCase` subclass https://github.com/rails/rails/blob/69256df843c40c6a8c9e9ecc91f4c59d7cbcd646/activesupport/lib/active_support/test_case.rb#L65